### PR TITLE
Cut 0.0.100

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Dependent on #1011, this cuts 0.0.100

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,15 +1,85 @@
-# 0.0.100 - WIP
+# 0.0.100 - 2021-08-???
+
+## API Updates
+ * rust-lightning can now be built in no_std mode, making it easy to target
```
Comment (valentinewallace) [inline CHANGELOG.md:4]: Our `*.md` files tend to use "Rust-Lightning" capitalization

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,15 +1,85 @@
...
+   field allowing you to select the maximum amount you are willing to pay to
+   avoid a force-closure. Further, we are now less restrictive on the fee
+   placed on the closing transaction when we are not the party paying it. To
+   control the feerate paid on a channel at close-time, use
+   `ChannelManager::close_channel_with_target_feerate` instead of
+   `close_channel` (#1011).
+ * `lightning_background_processor::BackgroundProcessor` now stops the
+   background thread when dropped (#1007). It is marked `#[must_use]` so that
+   Rust users will receive a compile-time warning when it is immediately
+   dropped after construction (#1029).
```
Comment (valentinewallace) [inline CHANGELOG.md:28]: How's this related to #1029?

**Review by valentinewallace — COMMENTED** [review]
ACK mod comments

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,15 +1,85 @@
...
 
 ## Serialization Compatibility
+ * Pending inbound keysend payments which have not either been failed or
+   claimed when serialized will result in a `ChannelManager` which is not
+   readable on pre-0.0.100 clients (#967).
+ * If
+   `lightning::chain::keysinterface::KeysInterface::get_shutdown_scriptpubkey`
+   returns a `ShutdownScript` which was constructed using a constructor other
+   than `ShutdownScript::new_p2wpkh_from_pubkey`, the `ChannelManager` will not
+   be readable by versions prior to 0.0.100. Further, versions prior to 0.0.100
```
Comment (jkczyz) [inline CHANGELOG.md:68]: That method is `pub(crate)` so you'll never see a `ShutdownScript` constructed that way in user code. So any new channels created will not be readable. However, any channel whose shutdown script was created using the old `get_shutdown_pubkey` API will continue to be readable.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,15 +1,85 @@
...
+   may not be able to read all `ChannelMonitorUpdate`s generated by 0.0.100
+   (#1019).
```
Comment (jkczyz) [inline CHANGELOG.md:70]: Maybe this should be phrased to describe when this is a problem (i.e. when `commit_upfront_shutdown_pubkey` is not set).

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,15 +1,85 @@
...
+ * LDK now supports sending and receiving "keysend" payments. This includes
+   modifications to `lightning::util::events::Event::PaymentReceived` to
+   indicate the type of payment (#967).
+ * A new variant, `lightning::util::events::Event::PaymentForwarded` has been
+   added which indicates a forwarded payment has been successfully claimed and
+   we've received a forwarding fee (#1004).
+ * `lightning::chain::keysinterface::KeysInterface::get_shutdown_pubkey` has
+   been renamed to `get_shutdown_scriptpubkey`, returns a script, and is only
+   called on shutdown if
+   `lightning::util::config::commit_upfront_shutdown_pubkey` is set (#1019).
```
Comment (jkczyz) [inline CHANGELOG.md:16]: Missing `ChannelConfig`

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,15 +1,85 @@
...
+   `ChannelManager::close_channel_with_target_feerate` instead of
+   `close_channel` (#1011).
+ * `lightning_background_processor::BackgroundProcessor` now stops the
+   background thread when dropped (#1007). It is marked `#[must_use]` so that
+   Rust users will receive a compile-time warning when it is immediately
+   dropped after construction (#1029).
+ * Total potential funds burn on force-close due to dust outputs is now limited
+   to `lightning::util::config::ChannelConfig::max_dust_htlc_exposure_msat` per
+   channel (#1009).
+ * The interval on which `lightning::ln::peer_handler::timer_tick_occurred`
```
Comment (jkczyz) [inline CHANGELOG.md:32]: Missing `PeerManager`

File: CHANGELOG.md (hunk 2 above)
Comment (TheBlueMatt) [reply CHANGELOG.md:28]: The must_use was added there, I can drop the mention if we prefer or just not reference where it was added

Comment (valentinewallace) [reply CHANGELOG.md:28]: Oh, right. Nw, resolving

File: CHANGELOG.md (hunk 3 above)
Comment (TheBlueMatt) [reply CHANGELOG.md:68]: Hmm, should we make that method pub?

**Comment by TheBlueMatt** [comment]
Tweaked changelog wordings to address feedback.

File: CHANGELOG.md (hunk 3 above)
Comment (jkczyz) [reply CHANGELOG.md]: Not sure. Actually, `KeysManager` is currently using`new_p2wpkh_from_pubkey`, so anyone using it for their `KeysInterface` shouldn't run into this problem.

File: CHANGELOG.md [hunk 7]
```diff
@@ -29,8 +30,9 @@
  * Total potential funds burn on force-close due to dust outputs is now limited
    to `lightning::util::config::ChannelConfig::max_dust_htlc_exposure_msat` per
    channel (#1009).
- * The interval on which `lightning::ln::peer_handler::timer_tick_occurred`
-   should be called has been reduced to once every five seconds (#1035) and
+ * The interval on which
+   `lightning::ln::peer_handler::PeerManagertimer_tick_occurred` should be
```
Comment (jkczyz) [inline CHANGELOG.md:34]: Missing `::`

**Review by jkczyz — APPROVED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,15 +1,94 @@
-# 0.0.100 - WIP
+# 0.0.100 - 2021-08-???
```
Comment (jkczyz) [inline CHANGELOG.md:1]: Update date.

**Review by valentinewallace — APPROVED** [review]

**Comment by TheBlueMatt** [comment]
Squashed without changes: `0b62f16b...11f78798`

**Pull request merged**